sdk/core/core-rest-pipeline/src/policies/tracingPolicy.ts

```diff
@@ -49,18 +45,14 @@ export function tracingPolicy(options: TracingPolicyOptions = {}): PipelinePolic
   return {
     name: tracingPolicyName,
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      if (!request.tracingOptions?.tracingContext) {
-        return next(request);
-      }
-
-      const span = tryCreateSpan(request, userAgent);
+      const { span, tracingContext } = tryCreateSpan(request, userAgent);
 
       if (!span) {
         return next(request);
       }
 
       try {
-        const response = await next(request);
+        const response = await tracingClient.withContext(tracingContext!, next, request);
         tryProcessResponse(span, response);
```

Comment:
if you are sure that `tracingContext` is defined, can you change its type when returned from `tryCreateSpan` to not allow undefined? If feels a bit risky to have the `!`

Reply:
Yeah that's totally fair. At runtime if span is defined tracingContext will also be defined (in the result of tryCreateSpan) but I think I model that much better - stay tuned! And thanks for the review